clients/src/main/java/org/apache/kafka/clients/consumer/internals/ListOffsetsRequestManager.java

```diff
@@ -158,13 +158,21 @@ public CompletableFuture<Map<TopicPartition, OffsetAndTimestamp>> fetchOffsets(
                 offsetFetcherUtils.buildOffsetsForTimesResult(timestampsToSearch, result.fetchedOffsets));
     }
 
+    /**
+     * Reset offsets for all assigned partitions that require it. Offsets will be reset
+     * with timestamps according to the reset strategy defined for each partition.
+     * <p>
+     * This may throw exception from previous offset fetch if there is one, ex.
```

Comment:
i see in the java doc it only throws: NoOffsetForPartitionException

is this correct?

Reply:
This may throw TopicAuthorizationException [here](https://github.com/philipnee/kafka/blob/f5a607910eb309422afad17e9fa5633989ea2bb1/clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcherUtils.java#L161), received as a response to the request. The processing that follows in case of a successful response is the one that may throw NoOffsetForPartitionException but that's at the KafkaConsumer level on the `subscriptions.resetInitializingPositions()`